tailnet/test/peer.go

```diff
@@ -180,6 +193,19 @@ func (p *Peer) AssertEventuallyGetsError(match string) {
 	}
 }
 
+// AssertNeverUpdateKind asserts that we have not received
+// any updates on the provided peer for the provided kind.
+func (p *Peer) AssertNeverUpdateKind(peer uuid.UUID, kind proto.CoordinateResponse_PeerUpdate_Kind) {
+	p.t.Helper()
+
+	updates, ok := p.peerUpdates[peer]
+	require.True(p.t, ok, "expected updates for peer %s", peer)
+
+	for _, update := range updates {
+		assert.NotEqual(p.t, kind, update.Kind, update)
```

Comment:
this function mixes between `require` and `assert` and I think it should stick to one or other.

Reply:
Since `require` is just shorthand for `t.Fatalf` and the rest of the function isn't relevant if we don't have updates we expect to be asserting against, I opted for it. The function is still technically correct if we just `assert` since we'll immediately fall out of the loop but imo semantically it makes more sense to return immediately.